### ISSUE TITLE
Upgrade Relay and the plugin to Babel 6

### ISSUE THREAD

**Issue by kassens** [facebook/relay#880]
We should upgrade the dependencies. This is currently blocked by missing `fbjs` transforms for Babel 6 that are used for the tests.

**Comment by kevinSuttle** [comment]
Yep. Breaks https://github.com/facebook/relay-starter-kit out of the box. 

```bash
⬡  npm start

> relay-starter-kit@0.1.0 start /Users/kevinSuttle/Code/relay-starter-kit
> babel-node ./server.js

/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/api/node.js:78
  throw new Error("The (" + alias + ") Babel 5 plugin is being run with Babel 6.");
  ^

Error: The (relay-query) Babel 5 plugin is being run with Babel 6.
    at new Plugin (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/api/node.js:78:9)
    at babelAdapter (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-relay-plugin/lib/babelAdapter.js:68:10)
    at /Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-relay-plugin/lib/getBabelRelayPlugin.js:50:12
    at Function.memoisePluginContainer (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:121:13)
    at Function.normalisePlugin (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:155:32)
    at /Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:195:30
    at Array.map (native)
    at Function.normalisePlugins (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:167:20)
    at OptionManager.mergeOptions (/Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:289:36)
    at /Users/kevinSuttle/Code/relay-starter-kit/node_modules/babel-core/lib/transformation/file/options/option-manager.js:309:17
```

**Comment by kevinSuttle** [comment]
As well as the tutorial at https://facebook.github.io/relay/docs/tutorial.html#content.

```bash
~/Code/relay-master/examples/relay-treasurehunt 6s
⬡  npm i

> babel-relay-plugin@0.7.3 prepublish /Users/kevinSuttle/Code/relay-master/scripts/babel-relay-plugin
> npm run build


> babel-relay-plugin@0.7.3 build /Users/kevinSuttle/Code/relay-master/scripts/babel-relay-plugin
> node scripts/build-lib

module.js:341
    throw err;
    ^

Error: Cannot find module 'babel-core'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/Users/kevinSuttle/Code/relay-master/scripts/babel-relay-plugin/scripts/build-lib:3:15)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
```

**Comment by kevinSuttle** [comment]
Looks like this has been addressed in the starter-kit.
https://github.com/relayjs/relay-starter-kit/commit/3ce3d579bcc4c3731fa6894bd493673c5ef8b8f0

**Comment by edvinerikson** [comment]
@kassens not sure if I'm missing something but Relay and babel-relay-plugin should be compiled with Babel 6 now? (checking if there's issues that should be closed)

**Comment by kassens** [comment]
This PR will help us upgrade the rest to Babel 6 more easily: facebook/fbjs#127

**Comment by kassens** [comment]
Fixed with #1109
